lib/relation-definition.js

```diff
@@ -1701,10 +1763,10 @@ HasOne.prototype.related = function (refresh, params) {
           cb(err);
         }
       });
-      return modelInstance[pk];
+      return cb.promise || modelInstance[pk];
```

Comment:
Again, `related()` should not support promise API, thus `cb.promise` is never set and this change can be reverted.